File: 0825-max-increase-to-keep-city-skyline/0825-max-increase-to-keep-city-skyline.cpp
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int n = grid.size();
        int ret = 0;

        vector<int> row(n, INT_MIN);
        vector<int> col(n, INT_MIN);

        for (int r = 0; r < n; r++){
            for (int c = 0; c < n; c++) {
                row[r] = max(row[r], grid[r][c]);
                col[c] = max(col[c], grid[r][c]);
            }
        }

        for (int r = 0; r < n; r++){
            for (int c = 0; c < n; c++) {
                ret+= min(row[r], col[c]) - grid[r][c];
            }
        }
        
        return ret;
    }
};

File: 0825-max-increase-to-keep-city-skyline/0825-max-increase-to-keep-city-skyline.py
class Solution:
    def maxIncreaseKeepingSkyline(self, grid: List[List[int]]) -> int:
        n: int = len(grid)
        ret: int = 0
        row: List[int] = [0] * n
        col: List[int] = [0] * n

        for r in range(n):
            for c in range(n):
                row[r] = max(row[r], grid[r][c])
                col[c] = max(col[c], grid[r][c])

        for r in range(n):
            for c in range(n):
                ret += min(row[r], col[c]) - grid[r][c]
        
        return ret